### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by HurnyakYaroslav**

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -0,0 +1 @@
+rootProject.name = 'permissionsAPI'
```
Comment (DmytroGromovyi) [inline settings.gradle]: change to service, it's not API

File: src/main/java/com/permissions/demo/configuration/SwaggerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@Configuration
+@EnableSwagger2
+public class SwaggerConfiguration {
+
+    @Bean
+    public Docket api() {
+        return new Docket(DocumentationType.SWAGGER_2)
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/configuration/SwaggerConfiguration.java]: Change to Open API 3, not Swagger 2

File: src/main/java/com/permissions/demo/configuration/SwaggerConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                .select()
+                .apis(RequestHandlerSelectors.withClassAnnotation(RestController.class))
+                .paths(PathSelectors.any())
+                .build()
+                .apiInfo(metaData());
+    }
+
+    private ApiInfo metaData() {
+        return new ApiInfoBuilder()
+                .title("Permissions-API")
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/configuration/SwaggerConfiguration.java]: Service

File: src/main/java/com/permissions/demo/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@RestController
+@AllArgsConstructor
+@RequestMapping("/user/")
+public class UserController {
+
+    private final UserService userService;
+
+    @GetMapping("/{userId}")
+    @ApiOperation(value = "Get user permissions info by user id")
+    public AuthSystemUserDTO get(@PathVariable Integer userId) {
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/controller/UserController.java]: Long

File: src/main/java/com/permissions/demo/dto/AuthSystemUserDTO.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class AuthSystemUserDTO {
+    private Integer userId;
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/dto/AuthSystemUserDTO.java]: Long

File: src/main/java/com/permissions/demo/model/AbstractEntity.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.MappedSuperclass;
+
+@MappedSuperclass
+@Data
+public class AbstractEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/model/AbstractEntity.java]: Long

File: src/main/java/com/permissions/demo/model/User.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import javax.persistence.Table;
+import java.util.List;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+@Entity
+@Table(name = "USER")
+public class User extends AbstractEntity {
+    private String name;
+    @ManyToMany
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/model/User.java:21]: One to Many

File: src/main/java/com/permissions/demo/repository/UserRepository.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.permissions.demo.repository;
+
+import com.permissions.demo.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Integer> {
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/repository/UserRepository.java]: Long

File: src/main/java/com/permissions/demo/service/UserService.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public AuthSystemUserDTO getUserById(Integer userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new RuntimeException("User not found with id: " + userId));
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/service/UserService.java]: UserNotFoundException extends Runtime, add GlobalControllerAdvice to handle such exception and return 404

File: src/main/java/com/permissions/demo/util/UserDTOMapper.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import com.permissions.demo.dto.AuthSystemUserDTO;
+import com.permissions.demo.dto.RoleDTO;
+import com.permissions.demo.model.Permission;
+import com.permissions.demo.model.Role;
+import com.permissions.demo.model.User;
+
+import java.util.List;
+
+
+public class UserDTOMapper {
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/util/UserDTOMapper.java]: Use Mapstruct implementation

File: src/main/resources/data.sql [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+insert into role(name)
+values ('ADMIN_ROLE'),
+       ('CUSTOM_ROLE'),
+       ('USER_ROLE');
+
+insert into role_permissions(role_id, permissions_id)
+values (1, 1),
+       (1, 2),
+       (1, 3),
```
Comment (DmytroGromovyi) [inline src/main/resources/data.sql:15]: Custom role, please add

File: src/main/java/com/permissions/demo/handler/GlobalControllerAdvice.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@RestControllerAdvice
+public class GlobalControllerAdvice extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(UserNotFoundException.class)
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/handler/GlobalControllerAdvice.java]: Exception class can be omited

File: src/main/java/com/permissions/demo/handler/GlobalControllerAdvice.java [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public String handleFileRequest(UserNotFoundException ex) {
```
Comment (DmytroGromovyi) [inline src/main/java/com/permissions/demo/handler/GlobalControllerAdvice.java]: rename to userNotFoundHandler

File: src/main/java/com/permissions/demo/model/User.java (hunk 7 above)
Comment (HurnyakYaroslav) [reply src/main/java/com/permissions/demo/model/User.java:21]: If we are considering a user with multiple roles, we must use the ManyToMany relationship.

**Pull request merged**